src/tests/base_test.py

```diff
@@ -383,27 +384,25 @@ def trigger_configurations(
     def controller(
         self, cluster_configuration: ClusterConfig, controller_configuration: BaseNodesConfig, trigger_configurations
     ) -> NodeController:
-        if cluster_configuration.platform == consts.Platforms.VSPHERE:
-            return VSphereController(controller_configuration, cluster_configuration)
-
-        if cluster_configuration.platform == consts.Platforms.NUTANIX:
-            return NutanixController(controller_configuration, cluster_configuration)
-
-        if cluster_configuration.platform == consts.Platforms.OCI:
-            return OciController(controller_configuration, cluster_configuration)
-
-        return TerraformController(controller_configuration, entity_config=cluster_configuration)
+        return self.get_terraform_controller(controller_configuration, cluster_configuration)
 
+    @classmethod
     def get_terraform_controller(
-        self, controller_configuration: BaseNodesConfig, cluster_configuration: ClusterConfig
-    ) -> TerraformController:
-        if cluster_configuration.platform == consts.Platforms.VSPHERE:
+        cls, controller_configuration: BaseNodesConfig, cluster_configuration: ClusterConfig
+    ) -> TerraformController | TFController:
```

Comment:
wait, we have ``TerraformController`` and ``TFController``? what's the difference between them?

Reply:
Yes, `TerraformController` is the one you familiar with.
A while back we added the OCI controller and I noticed that we duplicate a lot of code between the other node controllers (`VsphereController`, `NutanixController`, `OciController`), all of those with had a common base code that we moved to `TFController`.
The final goal is to create a `LibvirtController` that will also inherit from `TFController` and will do some order with the terraform/libvirt controllers.
You can see the description in this PR - https://github.com/openshift/assisted-test-infra/pull/2179

Reply:
The main problem with `TerraformController` are its dependencies, and moving the new base code will definitely will break QE (and probably our) tests. Keep it with backward compatibility will require a lot of work because the none/baremetal terraform files are not compatible with the new controller.

Reply:
yeah, I'm just commenting about the semantics, because:
* maybe we should have named it ``BaseTerraformController``? Currently it's hard seeing the relationship between the controller when reading their names
* Instead of ``TerraformController | TFController`` we can say this function returns ``TFController``, unless their API is very different from each other

Reply:
- I think we can rename it to `BaseTerraformController`, It's actually a better name I just didn't want for someone to think that `BaseTerraformController` is somehow related to `TerraformController`.
- I think they kind of different, the base interface (`NodeController`) is similar, but there are few other methods that are different. Furthermore it's actually a different object, so I don't think we should return `TFController` alone.

Anyway, Back then I successfully was able to move `baremetal` flow to the new base cluster but had some difficulties with `none` (uses different terraform files) so I left it aside for now